stream/src/main/java/com/annimon/stream/Stream.java

```diff
@@ -1670,6 +1670,63 @@ public boolean noneMatch(Predicate<? super T> predicate) {
     }
 
     /**
+     * Finds the first element and its index that matches the given predicate.
+     *
+     * <p>This is a short-circuiting terminal operation.
+     *
+     * <p>Example:
+     * <pre>
+     * predicate: (index, value) -&gt; index + value == 7
+     * stream: [1, 2, 3, 4, 5, 2, 0]
+     * index:  [0, 1, 2, 3, 4, 5, 6]
+     * result: Optional.of(IntPair(3, 4))
+     * </pre>
+     *
+     * @param predicate  the predicate to find value
+     * @return an {@code Optional} with {@code IntPair}
+     *         or {@code Optional.empty()} if stream is empty or no value was found.
+     * @since 1.1.8
+     */
+    public Optional<IntPair<T>> findIndexed(IndexedPredicate<? super T> predicate) {
```

Comment:
`Object` and `Number` are superclasses for `Integer`
So in the example above `?` is `Object` and `Number`, and `T` is Integer, which can be written as `class Integer extends Number` ->`class Number extends Object`
I understand.